common/djangoapps/util/milestones_helpers.py

```diff
@@ -86,33 +87,46 @@ def set_prerequisite_courses(course_key, prerequisite_course_keys):
             add_prerequisite_course(course_key, prerequisite_course_key)
 
 
-def get_pre_requisite_courses_not_completed(user, enrolled_courses):
+def get_pre_requisite_courses_not_completed(user, enrolled_courses):  # pylint: disable=invalid-name
     """
-    It would make dict of prerequisite courses not completed by user among courses
-    user has enrolled in. It calls the fulfilment api of milestones app and
-    iterates over all fulfilment milestones not achieved to make dict of
-    prerequisite courses yet to be completed.
+    Makes a dict mapping courses to their incomplete requirements using the
+    fulfillment API of the milestones app.
+
+    Arguments:
+        user (User): the user for whom we are checking prerequisites.
+        enrolled_courses (CourseKey): a list of keys for the courses to be
+            checked. The given user must be enrolled in all of these courses.
+
+    Returns:
+        dict[CourseKey: dict[
+            'courses': list[dict['key': CourseKey, 'display': str]]
+        ]]
+        If a course has no incomplete prerequisites, it will be excluded from the
```

Comment:
:ballot_box_with_check: (will push after you and Nimisha both finish your passes)